### ISSUE TITLE
Question about training-validation split implementation in PDBNABaseDataset

### ISSUE THREAD

**Issue by xiejz11113333** [rish-16/rna-backbone-design#10]
Hi @rish-16!
First, congratulations on the excellent work on RNA-FrameFlow! I'm a Master's student studying your paper and codebase for my research on RNA structure prediction.
I have a question regarding the training-validation split implementation in PDBNABaseDataset class (src/data/pdb_na_dataset_base.py).
Issue:
In the _init_metadata_and_splits() method, I noticed that when is_training=False (for validation set), the code performs:

```
eval_csv = pdb_csv  # Same data as training
eval_csv = eval_csv.groupby("modeled_na_seq_len").sample(
    self._data_conf.samples_per_eval_length, 
    replace=True,  # Allows sampling with replacement
    random_state=123
)
```

This appears to:
Use the same pdb_csv for both training and validation sets (validation data is a subset of training data)
Allow sampling with replacement (replace=True), meaning the same sample could appear multiple times in the validation set
From a machine learning perspective, this seems to create potential data leakage where the validation set is not independent from the training set.
Questions:
1.Was this design intentional? If so, could you explain the rationale? Is this a practical compromise given the limited availability of RNA structural data?
2.For the final evaluation in the paper, was a completely independent test set used? Or were the reported metrics based on this validation set?
3.The validation set only samples from num_eval_lengths length values - does this mean the validation metrics only represent performance on those specific lengths?
If I want to implement a standard train-val-test split for my own experiments, what would you recommend? Should I modify the replace=True to replace=False and ensure the validation set doesn't overlap with training?
Thank you for your time and for making this great work publicly available! Any clarification would be very helpful for my research.
Best regards
Xie Jingze
Master's Student,  China
